From ff79c2af89bb4a63d72133bd776e636f11a5dbb7 Mon Sep 17 00:00:00 2001
From: SupeChicken666 <me@supechicken666.dev>
Date: Fri, 18 Jul 2025 15:00:35 +0000
Subject: [PATCH 15/16] init: Import waydroid.prop last

We want to make sure that waydroid.prop is parsed last, so that it can override
any system properties.
That can only be done reliably by patching system/core/init

Change-Id: I04813cab789701db9e0afdee30d82ed11eab0d9b
Signed-off-by: SupeChicken666 <me@supechicken666.dev>
---
 init/property_service.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/init/property_service.cpp b/init/property_service.cpp
index 8cc95078ba..6f79d9fa0e 100644
--- a/init/property_service.cpp
+++ b/init/property_service.cpp
@@ -1214,6 +1214,7 @@ void PropertyLoadBootDefaults() {
     load_properties_from_file("/odm_dlkm/etc/build.prop", nullptr, &properties);
     load_properties_from_partition("odm", /* support_legacy_path_until */ 28);
     load_properties_from_partition("product", /* support_legacy_path_until */ 30);
+    load_properties_from_file("/vendor/waydroid.prop", nullptr, &properties);
  
     if (int vendor; ParseInt(properties["ro.vendor.build.version.sdk"], &vendor) &&
         vendor >= 33) {
-- 
2.50.1

